docs/posts/unix-backend/index.qmd

```diff
@@ -0,0 +1,255 @@
+---
+title: "Scaling to infinity and beyond: the Unix backend"
+author: "Phillip Cloud"
+date: "2024-04-01"
+categories:
+    - blog
+    - serious
+    - web-scale
+    - unix
+---
+
+## The Unix backend for Ibis
+
+We're happy to announce a new Ibis backend built on the world's best known web
+scale technology: Unix pipes.
+
+## Why?
+
+Why not? Pipes rock and they automatically stream data between operators and
+scale to your hard drive.
+
+What's not to love?
+
+## Demo
+
+All production ready backends ship with amazing demos.
+
+The Unix backend is no different. Let's see it in action.
+
+First we'll install the Unix backend.
+
+```bash
+pip install ibish
+```
+
+Like all production-ready libraries `ibish` depends on the latest commit of `ibis-framework`.
+
+Next we'll download some data.
+
+```{python}
+!curl -LsS 'https://storage.googleapis.com/ibis-examples/penguins/20240322T125036Z-9aae2/penguins.csv.gz' | zcat > penguins.csv
+```
+
+```{python}
+import ibis
+import ibish
+
+
+ibis.options.interactive = True
+
+unix = ibish.connect({"p": "penguins.csv"})
+
+t = unix.table("p")
+t
+```
+
+Sweet, huh?
+
+Let's filter the data and look at only the year 2009.
+
+```{python}
+expr = t.filter(t.year == 2009)
+expr
+```
+
+We can sort the result of that too, and filter again.
+
+```{python}
+expr = (
+    expr.order_by("species", ibis.desc("bill_length_mm"))
+    .filter(lambda t: t.island == "Biscoe")
+)
+expr
+```
+
+There's even support for joins and aggregations!
+
+Let's count the number of island, species pairs and sort descending by the count.
+
+```{python}
+expr = (
+    t.group_by("island", "species")
+    .agg(n=lambda t: t.count())
+    .order_by(ibis.desc("n"))
+)
+expr
+```
+
+For kicks, let's compare that to the DuckDB backend to make sure we're able to count stuff.
+
+To be extra awesome, we'll *reuse the same expression to do the computation*.
+
+```{python}
+ddb = ibis.duckdb.connect()
+ddb.read_csv("penguins.csv", table_name="p")  # <1>
+ddb.to_pandas(expr.unbind())
```

Comment:
The `to_pandas` makes the DuckDB output render differently than the `expr` output above, which makes it a little harder to visually confirm that they're identical.

Reply:
I'll wrap it in a memtable.

Reply:
Done.